cohttp-lwt-jsoo/src/cohttp_lwt_jsoo.ml

```diff
@@ -94,6 +98,32 @@ module Body_builder (P : Params) = struct
         else
           let u8a = new%js Typed_array.uint8Array_fromBuffer ab in
           CLB.of_string (string_of_uint8array u8a 0 ab##.byteLength)
+
+  let construct_body xml =
+    (* construct body *)
+    let b =
+      let respText () =
+        Js.Opt.case xml##.responseText
+          (fun () -> `String (Js.string ""))
+          (fun s -> `String s)
+      in
+      if xhr_response_supported then
```

Comment:
```ocaml
match  xhr_response_supported with
| true when Js.Opt.return xml##.response = Js.null -> 
  Firebug.console##log (Js.string "XHR Response is null; using empty string");
  `String (Js.string "")
| true ->
  Js.Opt.case
     (File.CoerceTo.arrayBuffer xml##.response)
     (fun () -> Firebug.console##log
                (Js.string
                   "XHR Response is not an arrayBuffer; using responseText");
              respText ())
            (fun ab -> `ArrayBuffer ab)
| -> respText ()
```
This is a readability suggestion, do you think it would be better or worse than the nested ifs?